numpy/core/shape_base.py

```diff
@@ -179,6 +179,86 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(arr, n, front=False):
+    r"""
+    View inputs as arrays with at least n dimensions.
+
+    Parameters
+    ----------
+    arr : array_like
+        One array-like object.  Non-array inputs are converted to arrays.
+        Arrays that already have n or more dimensions are preserved.
+
+    n : int
+        number of dimensions to ensure
+
+    front : bool
+        if True new dimensions are added to the front of the array.  otherwise
+        they are added to the back.
```

Comment:
This seems to contradict your implementation. "left" and "right" are probably clearer concepts here

Reply:
Unless my doctests are wrong (absolutely possible), I think its correct. 

```
"""
 >>> np.atleast_nd(x, 5, front=False).shape
(4, 3, 1, 1, 1)
 >>> np.atleast_nd(x, 5, front=True).shape
(1, 1, 1, 4, 3)
"""
```

If front is False, ones are appended at the end of shape (`shape[-1]`), and if its true they are appended at the start (`shape[0]`). Am I missing something?